services/templates/supabase-templates.service.ts

```diff
@@ -51,21 +51,44 @@ class SupabaseTemplateService {
     };
   }
 
-  async getTemplate(id: number): Promise<GenesisTemplate | null> {
+  async getTemplate(
+    id: number,
+    page: number = 1,
+    pageSize: number = 8,
+  ): Promise<PaginatedResponse<GenesisTemplate> | null> {
     const userData = await Service.require("oauth").getLoggedUser();
-
     const { data, error } = await this.supabase
       .from("templates")
       .select("*")
       .eq("id", id)
       .eq("ownerId", userData?.id)
-      .select();
+      .single();
+
     if (error) {
       messageHandler.handleError(error.message);
       return null;
     }
 
-    return data[0] as GenesisTemplate;
+    const entries = Object.entries(data.fields);
+    const totalCount = entries.length;
+    const totalPages = Math.ceil(totalCount / pageSize);
+
+    const paginatedEntries = entries.slice(
+      (page - 1) * pageSize,
+      page * pageSize,
+    );
```

Comment:
This logic needs to be in the table/pagination.

Most APIs are paginated, Does sense we always do the same calculation? 

API required: records, total count.
Table knows which page is selected and the items per page. 

Conclusion: this formula is only once , in once place, for all the type of APIs.

Reply:
If this is stable and tested create a task for that and we do after.
Add a `//TODO ` to know we need to refactor with my comment